src/components/course/data/utils.jsx

```diff
@@ -407,10 +409,344 @@ export function shortenString(str, maxLength, suffix, separator = ' ') {
+    DISABLED_ENROLL_REASON_TYPES.SUBSCRIPTION_LICENSE_NOT_ASSIGNED,
+    DISABLED_ENROLL_REASON_TYPES.COUPON_CODE_NOT_ASSIGNED,
+    DISABLED_ENROLL_REASON_TYPES.COUPON_CODES_EXPIRED,
+  ].includes(reasonType);
+
+  if (hasLimitsLearnMoreCTA) {
+    ensureConfig(['LEARNER_SUPPORT_SPEND_ENROLLMENT_LIMITS_URL']);
+    return (
+      <Button
+        as={Hyperlink}
+        className="text-center"
+        destination={getConfig().LEARNER_SUPPORT_SPEND_ENROLLMENT_LIMITS_URL}
+        target="_blank"
+        size="sm"
+        block
+      >
+        Learn about limits
+      </Button>
+    );
+  }
+
+  if (hasDeactivationLearnMoreCTA) {
+    ensureConfig(['LEARNER_SUPPORT_ABOUT_DEACTIVATION_URL']);
+    return (
+      <Button
+        as={Hyperlink}
+        className="text-center"
+        destination={getConfig().LEARNER_SUPPORT_ABOUT_DEACTIVATION_URL}
+        target="_blank"
+        size="sm"
+        block
+      >
+        Learn about deactivation
+      </Button>
+    );
+  }
+
+  if (hasContactAdministratorCTA) {
+    if (enterpriseAdminUsers?.length === 0) {
+      return null;
+    }
+    const adminEmails = enterpriseAdminUsers.map(({ email }) => email).join(',');
+    return (
+      <Button
+        as={MailtoLink}
+        className="text-center"
+        to={adminEmails}
+        target="_blank"
+        size="sm"
+        block
+      >
+        Contact administrator
+        <span className="sr-only">for help</span>
+      </Button>
+    );
+  }
+
+  return null;
+};
+
+export const getMissingApplicableSubsidyReason = ({
+  enterpriseAdminUsers,
+  catalogsWithCourse,
+  couponCodes,
+  couponsOverview,
+  customerAgreementConfig,
+  subscriptionLicense,
+  containsContentItems,
+  missingSubsidyAccessPolicyReason,
+}) => {
+  // Default disabled enrollment reason, assumes enterprise customer does not have any administrator users.
+  let reasonType = DISABLED_ENROLL_REASON_TYPES.NO_SUBSIDY_NO_ADMINS;
+  let userMessage;
+  const hasEnterpriseAdminUsers = enterpriseAdminUsers?.length > 0;
+
+  // If there are admin users, change `reasonType` to use the
+  // `DISABLED_ENROLL_REASON_TYPES.NO_SUBSIDY` message.
+  if (hasEnterpriseAdminUsers) {
+    reasonType = DISABLED_ENROLL_REASON_TYPES.NO_SUBSIDY;
+  }
+
+  const couponCodesDisabledEnrollmentReasonType = getCouponCodesDisabledEnrollmentReasonType({
+    catalogsWithCourse,
+    couponCodes,
+    couponsOverview,
+    hasEnterpriseAdminUsers,
+  });
+  const subscriptionsDisabledEnrollmentReasonType = getSubscriptionDisabledEnrollmentReasonType({
+    customerAgreementConfig,
+    catalogsWithCourse,
+    subscriptionLicense,
+    hasEnterpriseAdminUsers,
+  });
+  const enterpriseOffersDisabledEnrollmentReasonType = getEnterpriseOffersDisabledEnrollmentReasonType({
+    enterpriseOffers: customerAgreementConfig?.enterpriseOffers,
+    catalogsWithCourse,
+  });
+  const learnerCreditDisabledEnrollmentReasonType = getLearnerCreditDisabledEnrollResponse({
```

Comment:
[clarification] Is this `getLearnerCreditDisabledEnrollResponse` function necessary given the reason type and its associated user message is already known based on the `missingSubsidyAccessPolicyReason` argument passed to this function?

We only need to parse the legacy subsidies to determine a disabled enroll reason (i.e., subs, offers, codes) because the `can-redeem` API returns the reason type and user message without needing the MFE to determine the reason type on its own.

That is, we should rely on the reason type and user message given to the function via the `missingSubsidyAccessPolicyReason` function argument, which should contain a data structure similar to the following ([source](https://enterprise-access.edx.org/api/schema/redoc/#tag/Subsidy-Access-Policy-Redemption/operation/api_v1_policy_redemption_enterprise_customer_can_redeem_list)):

```json
{
  "reason": "string",
  "user_message": "string",
  "policy_uuids": [],
  "metadata": {}
}
```